tests/PHPStan/Rules/Methods/CallStaticMethodsRuleTest.php

```diff
@@ -445,6 +445,28 @@ public function testBug5259(): void
 		$this->analyse([__DIR__ . '/data/bug-5259.php'], []);
 	}
 
+	public function testBug5782(): void
+	{
+		$this->checkThisOnly = false;
+		$this->analyse([__DIR__ . '/data/bug-5782.php'], [
+			[
+				'Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{\'Bug5782\\\HelloWorld\', \'sayGoodbye\'} given.',
+				23,
+			],
+			[
+				'Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{\'Bug5782\\\HelloWorld\', \'sayGoodbye\'} given.',
+				30,
+			],
+			// @todo this should also error on mixed type, but doesn't?
+			// ConstantStringType::getCallableParametersAcceptors returns a
+			// MixedType but that doesn't cause an error like ConstantArrayType.
+			[
+				'Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{\'Bug5782\\\HelloWorld\', \'sayGoodbye\'} given.',
+				31,
+			],
```

Comment:
I believe adding the scope to `ConstantStringType::isCallable` will also cause this test failure to be fixed. Without the scope to know if we're in a class or not, this always returns as `yes` for the method being callable when it actually may not be.

```
			$classRef = $reflectionProvider->getClass($matches[1]);
			if ($classRef->hasMethod($matches[2])) {
				return TrinaryLogic::createYes();
			}
```